geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/sortedset/AbstractZInterStoreIntegrationTest.java

```diff
@@ -78,6 +78,32 @@ public void shouldError_givenWrongKeyType() {
         STRING_KEY, KEY1)).hasMessage("WRONGTYPE " + RedisConstants.ERROR_WRONG_TYPE);
   }
 
+  @Test
+  public void shouldError_givenTwoSortedSetKeysAndWrongKeyType() {
+    Map<String, Double> scores = buildMapOfMembersAndScores(1, 5);
+    Map<String, Double> nonIntersectionScores = buildMapOfMembersAndScores(6, 10);
+    jedis.zadd(KEY1, scores);
+    jedis.zadd(KEY2, nonIntersectionScores);
```

Comment:
The set-up in this test can be simplified, since we don't really care what's in the sorted sets, just that they ARE sorted sets. Something like this maybe:
```
    jedis.zadd(KEY1, 1, "value1");
    jedis.zadd(KEY2, 1, "value2");
```